src/test/article.spec.js

```diff
@@ -0,0 +1,158 @@
+
+  beforeEach(async () => {
+    article = await BaseRepository.create(db.Article, articleSample);
+  });
+  afterEach(async () => {
+    await db.Article.destroy({ truncate: true, cascade: true });
+  });
+
+  describe('CREATE ARTICLE', () => {
+    it('should create article successfully, with valid user input', done => {
+      chai
+        .request(app)
+        .post(articleEndpoint)
+        .set('x-access-token', token)
+        .send(articleSample)
+        .end((error, response) => {
+          expect(response.status).to.equal(201);
+          done();
+        });
+    });
+  });
+
+  describe('GET ALL ARTICLES', () => {
+    before(() => {});
+
+    it('should fetch all successfully created articles', done => {
+      chai
+        .request(app)
+        .get(articleEndpoint)
+        .set('x-access-token', token)
+        .end((error, response) => {
+          expect(response.status).to.equal(200);
+          expect(response.body.data.length).to.be.greaterThan(0);
+          expect(response.body.data[0]).to.have.property('id');
+          expect(response.body.data[0]).to.have.property('title');
+          expect(response.body.data[0]).to.have.property('description');
+          expect(response.body.data[0]).to.have.property('content');
+          expect(response.body.data[0]).to.have.property('image');
+          expect(response.body.data[0]).to.have.property('slug');
+          expect(response.body.data[0]).to.have.property('authorId');
+          expect(response.body.data[0]).to.have.property('status');
+          expect(response.body.data[0]).to.have.property('publisheddate');
+          expect(response.body.data[0]).to.have.property('createdAt');
+          expect(response.body.data[0]).to.have.property('updatedAt');
+          done();
+        });
+    });
+  });
+
+  describe('GET A SPECIFIC ARTICLE', () => {
+    it('should successfully get a specific', done => {
+      chai
+        .request(app)
+        .get(`${articleEndpoint}/${article.id}`)
+        .set('x-access-token', token)
+        .send()
+        .end((error, response) => {
+          expect(response.body.status).to.equal(200);
+          done();
+        });
+    });
+
+    it('should throw a 404 error when article is not found', done => {
+      chai
+        .request(app)
+        .get(`${articleEndpoint}/345`)
+        .set('x-access-token', token)
+        .send()
+        .end((error, response) => {
+          expect(response.status).to.equal(404);
+          expect(response.body.message).to.equal('Article Not Found');
+          done();
+        });
+    });
+  });
+
+  describe('UPDATE AN ARTICLES', () => {
+    before(() => {});
+
+    it('should successfully update a specific article', done => {
+      chai
+        .request(app)
```

Comment:
a `PUT` request requires that you send the whole resource, that is an article, with updated fields. So, if the request doesn't have all the valid fields like a `create`, it should fail. There should also be an article validation for the individual fields.

See @timi-codes and @kechyy for this.

Reply:
Thank you, work is in progress to implement the feedback received on this review.